glusterd2/xlator/options/options.go

```diff
@@ -212,23 +243,33 @@ func ValidateSizeList(o *Option, val string) error {
 
 // ValidateSizet validates if the option is a valid size
 func ValidateSizet(o *Option, val string) error {
-	return ErrInvalidArg
+	v, err := strconv.ParseFloat(val, 64)
+	if err != nil {
+		return err
+	}
+	if o.Min != 0 && o.Max != 0 {
+		return nil
+	} else if v < o.Min || v > o.Max {
+		return ErrInvalidRange
+	}
+	return nil
 }
 
 // ValidateStr validates if the option is of type Str
 func ValidateStr(o *Option, val string) error {
-	l := len(val)
+	l := len(strings.TrimSpace(val))
```

Comment:
Save the trimmed string, and use it when checking against expected values.

Reply:
eeks! an overlook, am sorry.